packages/framework-provider-azure-infrastructure/src/infrastructure/index.ts

```diff
@@ -1,50 +1,42 @@
+import { azureCredentials, createResourceGroupName, createResourceManagementClient } from './helper/utils'
+import { runCommand } from '@boostercloud/framework-common-helpers'
+import { InfrastructureRocket } from './rockets/infrastructure-rocket'
+import { ApplicationBuilder } from './application-builder'
+import { RocketBuilder } from './rockets/rocket-builder'
 
-export const synth = (configuration: BoosterConfig, logger: Logger): Promise<void> => synthApp(logger, configuration)
+export const synth = (configuration: BoosterConfig, logger: Logger, rockets?: InfrastructureRocket[]): Promise<void> =>
+  synthApp(logger, configuration, rockets)
 
-export const deploy = (configuration: BoosterConfig, logger: Logger): Promise<void> => deployApp(logger, configuration)
+export const deploy = (configuration: BoosterConfig, logger: Logger, rockets?: InfrastructureRocket[]): Promise<void> =>
+  deployApp(logger, configuration, rockets)
 
 export const nuke = (configuration: BoosterConfig, logger: Logger): Promise<void> => nukeApp(logger, configuration)
 
 /**
  * Synth the application for Azure
  */
-async function synthApp(logger: Logger, config: BoosterConfig): Promise<void> {
-  await terraformSynth(logger, config)
-}
-
-async function uploadFile(logger: Logger, config: BoosterConfig, zipResource: ZipResource): Promise<void> {
-  logger.info('Uploading zip file')
-  const resourceGroupName = createResourceGroupName(config.appName, config.environmentName)
-  const functionAppName = createFunctionResourceGroupName(resourceGroupName)
-  await FunctionZip.deployZip(functionAppName, resourceGroupName, zipResource)
-  logger.info('Zip file uploaded')
+async function synthApp(logger: Logger, config: BoosterConfig, rockets?: InfrastructureRocket[]): Promise<void> {
+  const applicationBuilder = new ApplicationBuilder(logger, config, rockets)
+  await applicationBuilder.buildApplication()
 }
 
 /**
  * Deploys the application in Azure
  */
-async function deployApp(logger: Logger, config: BoosterConfig): Promise<void> {
+async function deployApp(logger: Logger, config: BoosterConfig, rockets?: InfrastructureRocket[]): Promise<void> {
   logger.info(`Deploying app ${config.appName}`)
-  const zipResource = await terraformSynth(logger, config)
+  const applicationBuilder = new ApplicationBuilder(logger, config, rockets)
+  const applicationBuild = await applicationBuilder.buildApplication()
+
   const command = await runCommand(process.cwd(), 'npx cdktf deploy --auto-approve')
   if (command.childProcess.exitCode !== 0) {
     return Promise.reject(`Deploy application ${config.appName} failed. Check cdktf logs`)
   }
-  await uploadFile(logger, config, zipResource)
+
+  await applicationBuilder.uploadFile(applicationBuild.zipResource)
+  const rocketBuilder = new RocketBuilder(logger, config, applicationBuild.azureStack.applicationStack, rockets)
+  rocketBuilder.uploadRocketsFiles(applicationBuild.rocketsZipResources)
```

Comment:
Question: Do the rockets generate different templates? (asking it because I see we are uploading them separately).

I thought that, in the end, we have only one template that the rockets have modified.

Reply:
Not really. When you have rockets in your project only one template will be generated with all the infra needed. 

What we will have is a zip file for each rocket (and one for the app). The zip files are the FunctionsApp code that needs to be uploaded using the Azure cli tool to each FunctionApp. This is needed because we are not deploying the code using Terraform yet.